presto-hive/src/main/java/com/facebook/presto/hive/HivePartitioningHandle.java

```diff
@@ -49,21 +81,33 @@ public int getBucketCount()
     }
 
     @JsonProperty
-    public List<HiveType> getHiveTypes()
+    public Optional<List<HiveType>> getHiveTypes()
     {
         return hiveTypes;
     }
 
+    @JsonProperty
+    public Optional<List<Type>> getTypes()
+    {
+        return types;
+    }
+
     @JsonProperty
     public OptionalInt getMaxCompatibleBucketCount()
     {
         return maxCompatibleBucketCount;
     }
 
+    @JsonProperty
+    public HiveBucketProperty.BucketingType getBucketingType()
```

Comment:
Let's extract `BucketingType` into a standalone class. Let's call it `BucketFunctionType`.